src/main/webapp/js/login.js

```diff
@@ -0,0 +1,384 @@
+        console.log('User signed out.');
+    });
+    localStorage.setItem("user", 0);
+    hideAccountMenu();
+}
+
+/*
+    FUNCTIONS RELATED TO THE ACCOUNTS PAGE
+*/
+
+function accountFunctions() {
+    getNumSearches();
+    getLastVisited();
+    getFavFood();
+    getNumReviews();
+}
+
+function getNumSearches() {
+    let count = 0;
+    let numSearchesEl = document.getElementById('num-searches');
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    fetch(`/searches?user=${userID}`, { method: 'GET' }).then(response => response.json()).then((searches) => {
+        searches.forEach((search) => {
+            count += 1;
+        });
+        numSearchesEl.innerText = count;
+    });
+}
+
+function getLastVisited() {
+    let lastVisitedEl = document.getElementById('last-visited');
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    fetch(`/searches?user=${userID}`, { method: 'GET' }).then(response => response.json()).then((searches) => {
+        for (search of searches) {
+            lastVisitedEl.innerText = search.name;
+            break;
+        }
+    });
+}
+
+function getFavFood() {
+    let foodHolder = document.getElementById('fav-food');
+    fetch(`/fav-food?user=${userID}`, { method: 'GET' }).then(response => response.json()).then((foods) => {
+        if (foods.length == 0) {
+            console.log("hello");
+            let favFoodFormEl = document.createElement('form');
+            favFoodFormEl.action = "/fav-food";
+            favFoodFormEl.method = "POST";
+            favFoodFormEl.id = "fav-food-form";
+
+            let favFoodInputEl = document.createElement('textarea');
+            favFoodInputEl.id = "food-selection";
+            favFoodInputEl.placeholder = "or foods :)";
+
+            let formInputHolderEl = document.createElement('div');
+            formInputHolderEl.class = "input-group";
+
+            let inputButtonEl = document.createElement('input');
+            inputButtonEl.type = "submit";
+
+            formInputHolderEl.appendChild(favFoodInputEl);
+            favFoodFormEl.appendChild(formInputHolderEl);
+            favFoodFormEl.appendChild(inputButtonEl);
+            foodHolder.appendChild(favFoodFormEl);
+        } else {
+            for (food of foods) {
+                let foodTextEl = document.createElement('p');
+                foodTextEl.innerText = food;
+                foodHolder.appendChild(foodTextEl);
+                break;
+            }
+        }
+    });
+}
+
+function getNumReviews() {
+    let count = 0;
+    let numFeedbackEl = document.getElementById('num-feedback');
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    fetch(`/feedback?user=${userID}`, { method: 'GET' }).then(response => response.json()).then((feedbackList) => {
+        feedbackList.forEach((feedback) => {
```

Comment:
is there a way to get the size of the list instead of counting through it?

Reply:
done